### PR TITLE
Massive changeling overhaul holy shit

### PR DESCRIPTION
**Pull request by Crystalwarrior**
Thanks to @Xhuis for the original PR ( https://github.com/tgstation/-tg-station/pull/13138 )!

Current tier costs
^Above image shows all cost changes and absorbtion requirements to the changeling abilities
- experimental: "Complete changeling rework! Now changeling abilities are divided into tiers based on required amount of absorbtions to unlock them. You start with 5 Evolution Points by default and have access to Tier 1, which includes various stings. You also have lesser form and mimic voice abilities from the start."
- experimental: "This is a VERY experimental update, and all player feedback is much appreciated!"
- rscremove: "Removed Changeling Shield, Chitinous Armor and Extract DNA sting"
- rscremove: "Removed Changeling team objectives"
- rscadd: "Added synaptic enhancement as a changeling ability. When active, this will halve the delay between actions such as attacking."
- rscadd: "Paralysis Sting has been re-added and revamped. After a short delay based on the victim's health, they will be knocked down for fifteen seconds."
- rscadd: "Death Sting has been re-added and revamped. It will inject them with a small dosage of Wasting Toxin with slow metabolization rate, which will slowly drain blood and deal toxin damage to the victim."
- rscadd: "Horror Form has been added. When used, this will transform the changeling into their powerful true form, able to deal powerful damage, crawl through vents, and tear into living and dead humans to heal themselves. They will be incapable of returning to a human for three minutes, and when they do, are stunned for ten seconds. Death will automatically turn them back into humans and knock them out for thirty seconds."
- tweak: "Chameleon ability for Changelings now makes them invisible to mouse cursor when they're completely invisible."
- tweak: "Chameleon ability has been sped up slightly."
- tweak: "Evolution points have received a large rebalance. Changelings now start with five points, and will gain two more after each absorption. Some abilities take more than five points to unlock, meaning powerful abilities like Fleshmend will be locked by default. If a changeling absorbs another changeling, the absorber will gain the other's remaining genetic points."
- tweak: "There is now no limit on genomes storable for transformation. This means that you will be able to transform into anyone you have absorbed past the first six."
- tweak: "Mute Sting now functions based on the Mute Toxin reagent and lasts for much shorter."
- tweak: "Blind Sting now confuses the victim for a short duration."
- tweak: "All stings but transformation sting no longer function on the dead. Cryogenic Sting now only works on conscious targets."
- bugfix: "Fixed transformations sometimes giving you 'fake' items that prevented you from equipping the real deal"
- bugfix: "Fixed lesser form dropping ling eye implants"

### CODE REVIEW

**Comment by danhibiki** [comment]
:+1:

File: code/game/gamemodes/changeling/changeling.dm [hunk 1]
```diff
@@ -421,12 +423,12 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
 	user.underwear = chosen_prof.underwear
 	user.undershirt = chosen_prof.undershirt
 	user.socks = chosen_prof.socks
-	user.dna = chosen_dna //Should fix the whole "ERM MEH GERD I TURNED INTO A HUMAN INSTEAD OF ROBIT"
```
Comment (TDcoolguy300) [inline code/game/gamemodes/changeling/changeling.dm:424]: We've got an issue here Crystal, that was there to deal with changlings not getting the correct species, unless you managed to handle it elsewhere now.

File: code/game/gamemodes/changeling/changeling.dm [hunk 2]
```diff
@@ -421,12 +423,12 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
...
+
 	chosen_dna.transfer_identity(user, 1)
 	user.updateappearance(mutcolor_update=1)
 	user.update_body()
 	user.domutcheck()
-/* //Seriously, why was this even added?
```
Comment (TDcoolguy300) [inline code/game/gamemodes/changeling/changeling.dm:429]: Wasnt this suppose to be commented out?

**Comment by Crystalwarrior** [comment]
I feel this is merge-ready now.
We obviously need to test the FUCK out of this update, preferably live because my test server sucks dicks.

**Comment by Chronitonity** [comment]
Write up what this changes, removes, adds, ect in your OP please.

**Comment by Crystalwarrior** [comment]
gimme a sec

**Comment by FuryMcFlurry** [comment]
I'll be doing sprites, reworking the UI n what not

File: code/game/gamemodes/changeling/changeling.dm (hunk 2 above)
Comment (Crystalwarrior) [inline code/game/gamemodes/changeling/changeling.dm:429]: I re-commented it out again, hue

**Comment by FuryMcFlurry** [comment]
stingers

today I learned coders cannot sprite for shit.

**Comment by Chronitonity** [comment]
@FuryMcFlurry jump on the server and PM me your email.

**Comment by Spacedong** [comment]
this is ebin

**Comment by calibraptor** [comment]
This is pretty noice.

**Comment by Crystalwarrior** [comment]
So, when can we test this live?

**Comment by Chronitonity** [comment]
http://puu.sh/lEc7B/ba7838b46c.png

**Pull request merged**